### DISCUSSION TITLE
Overwrite the default "Not Found" HTTPException detail message globally

### DISCUSSION THREAD

**Discussion by xiki808** [fastapi/fastapi#6511] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
No code needed.
```

### Description

As the title says, I'm trying to change the default "Not Found" text for the detail key when a URL that does not exist is requested. How can I do this?

### Operating System

Linux

### FastAPI Version

0.73

### Python Version

3.9

**Comment by jgould22** [comment]
You can likely just pass in your own like you would with Starlette  but I have not tried it however.

https://www.starlette.io/exceptions/

**Comment by xiki808** [comment]
@jgould22 Thank you, I tried something similar but couldn't get it right. That document helped me a bit more.

```
from fastapi.responses import JSONResponse

from .const import (
    REQUEST_NOT_FOUND
)

async def not_found(request, exc):
    return JSONResponse(content={'detail': REQUEST_NOT_FOUND}, status_code=exc.status_code)

exception_handlers = {
    404: not_found
}

app = FastAPI(exception_handlers=exception_handlers)

```

**Comment by Kludex** [comment, marked as answer]
```python
from fastapi.responses import JSONResponse
from fastapi import FastAPI


async def not_found(request, exc):
    return JSONResponse(content={'detail': "Not found haha"}, status_code=exc.status_code)

exception_handlers = {
    404: not_found
}

app = FastAPI(exception_handlers=exception_handlers)
```

Works just fine.
